crypto/bio/bio_lib.c

```diff
@@ -848,29 +857,20 @@ int BIO_connect_retry(BIO *bio, int timeout)
         return 0;
     }
 
+    if (polling_milliseconds < 0)
+        polling_milliseconds = 100;
     if (!blocking)
         BIO_set_nbio(bio, 1);
 
- retry: /* it does not help here to set SSL_MODE_AUTO_RETRY */
-    rv = BIO_do_handshake(bio); /* This indirectly calls ERR_clear_error(); */
+ retry: /* it does not always help to set SSL_MODE_AUTO_RETRY */
+    clear_sys_error();
+    rv = BIO_do_connect(bio); /* This indirectly calls ERR_clear_error(); */
```

Comment:
looking at It bss_conn.c it don't look like always calling ERR_clear_error,
I only see that happening when the next address is tried (blocking mode)
or when a non-blocking connect is started.

Reply:
Correct. Yet my comment was not meant as an assertion that `ERR_clear_error()` is called in any case but as a warning that it may be called, such that any previous error queue contents may be lost.
I've meanwhile made this clear, also in the documentation, in my today's fixup commit.